subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -134,6 +137,20 @@ public String getTestFramework() {
         return testFramework;
     }
 
+    /**
+     * How to handle insecure (http) URLs used for Maven Repositories.
+     *
+     * This property can be set via command-line option '--insecure-protocols'.
+     *
+     * @since 7.3
+     */
+    @Nullable
+    @Input
+    @Optional
+    public String getInsecureProtocolHandler() {
```

Comment:
We should make this a `Property<String>` and that lets us remove the `@Nullable` and `@Optional` annotations and the setter below. The `@Option` annotation should then move up to here. 

In `BuildInitPlugin`, we should then set the convention value to "warn"

Since `InitBuild` is not abstract, you'll have to add a field of type `Property<String>` and initialize it via ObjectFactory.

I would also go with the property name `insecureProtocol` to match the command-line option